backend/dataall/modules/s3_datasets_shares/db/share_object_repositories.py

```diff
@@ -683,6 +683,24 @@ def get_share_items_states(session, share_uri, item_uris=None):
             query = query.filter(ShareObjectItem.shareItemUri.in_(item_uris))
         return [item.status for item in query.distinct(ShareObjectItem.status)]
 
+    @staticmethod
+    def get_share_items_health_states(session, share_uri, item_uris=None):
+        query = (
+            session.query(ShareObjectItem)
+            .join(
+                ShareObject,
```

Comment:
Do we need the join with the ShareObject? I think we can filter directly on ShareObjectItem.shareUri == share_uri

Reply:
removed join - good call out